cms/djangoapps/contentstore/views/item.py

```diff
@@ -266,6 +269,8 @@ def service(self, block, service_name):
                 return DjangoXBlockUserService(self._user)
             if service_name == "studio_user_permissions":
                 return StudioPermissionsService(self._user)
+            if service_name == "lti-configuration":
+                return ConfigurationService(LTIConsumerHideFieldsFlag)
```

Comment:
Here is the PR where this was added: #11331. @doctoryes, @bradenmacdonald or @dmitry-viskov might be able to shed more light.

Looking at the code, it appears that a different runtime is used for editing XBlocks vs. previewing them in Studio (and `StudioPermissionsService` is only needed when editing). I think that makes sense, but I don't know if we have other paradigms such as adding services to an existing runtime. @cpennington can you comment on how/if runtimes are intended to be swapped out?